### ISSUE TITLE
Wazuh try to upgrade a Centos5 Agent despite there is no WPK for Centos5 system. 

### ISSUE THREAD

**Issue by MiguelCasaresRobles** [wazuh/wazuh#3387]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
|All versions | WPK | Manager/Agent | Packages/Sources | Centos5-Agent/All-Manager |

Hello Team,

I have realized that when we try to upgrade a Centos 5 Agents from the Manager using a WPK, the manager does not refuse the upgrade despite there is no WPK for Centos 5.

This is the debug output of the `agent_upgrade` tool:

```
[root@localhost vagrant]# /var/ossec/bin/agent_upgrade -a 007 -v v3.9.0 -d
Manager version: v3.9.0
Agent version: v3.8.2
Agent new version: v3.9.0
WPK file already downloaded: /var/ossec/var/upgrade/wazuh_agent_v3.9.0_linux_x86_64.wpk - SHA1SUM: 1
Upgrade PKG: wazuh_agent_v3.9.0_linux_x86_64.wpk (4492.53125 KB)
MSG SENT: 007 com open wb wazuh_agent_v3.9.0_linux_x86_64.wpk
RESPONSE: ok
MSG SENT: 007 com lock_restart -1
RESPONSE: ok
Chunk size: 512 bytes
Sending: /var/ossec/var/upgrade/wazuh_agent_v3.9.0_linux_x86_64.wpk
MSG SENT: 007 com close wazuh_agent_v3.9.0_linux_x86_64.wpk
RESPONSE: ok
MSG SENT: 007 com sha1 wazuh_agent_v3.9.0_linux_x86_64.wpk
RESPONSE: ok 180e25a1fefafe8d83c763d375cb1a3a387bc08a
WPK file sent
MSG SENT: 007 com upgrade wazuh_agent_v3.9.0_linux_x86_64.wpk upgrade.sh
RESPONSE: ok 0
Upgrade procedure started
Error 1716: Error upgrading agent: Timeout waiting for agent reconnection.
Traceback (most recent call last):
  File "/var/ossec/framework/scripts/agent_upgrade.py", line 170, in <module>
    main()
  File "/var/ossec/framework/scripts/agent_upgrade.py", line 139, in main
    raise WazuhException(1716, "Timeout waiting for agent reconnection.")
wazuh.exception.WazuhException: Error 1716 - Error upgrading agent: Timeout waiting for agent reconnection.
```

I have attached the `upgrade.log` of the Agent:

[upgrade.log.txt](https://github.com/wazuh/wazuh/files/3212109/upgrade.log.txt)

The most important thing is after failed upgrading, the WPK changes the permissions of the `/etc` folder from `root:root` to `root:ossec` and the same for the `wazuh-agent` file under `/etc/rc` folders.

Regards,

Miguel Casares

**Comment by chemamartinez** [comment]
Thank you @MiguelCasaresRobles for reporting this issue. It has been already fixed at #3391.

The permissions issue is provoked by installing an incompatible version of the Wazuh agent on CentOS 5.

Regards.

**Comment by chemamartinez** [comment]
Merged at #3391
